Cloudinary/Actions/BaseResult.cs

```diff
@@ -120,11 +120,13 @@ public abstract class BaseResult
         /// </summary>
         /// <param name="response">HTTP response</param>
         /// <returns>New instance of this class</returns>
-        internal static T Parse<T>(HttpWebResponse response) where T : BaseResult, new()
+        internal static T Parse<T>(Object responseObj) where T : BaseResult, new()
         {
-            if (response == null)
+            if (responseObj == null)
                 throw new ArgumentNullException("response");
 
+            HttpWebResponse response = (HttpWebResponse)responseObj;

```

Comment:
This line will throw an exception if responseObj is not of type HttpWebResponse. The previous implementation was type safe and now we lost this "safety".
Other option is to move this line before the IF in line 125 and change it to be HttpWebResponse response = responseObj as HttpWebResponse; this will at least won't throw an exception (and you can write to log if there is an issue with the type of the argument)